payu/models/access.py

```diff
@@ -157,20 +176,26 @@ def setup(self):
                         self.expt.runtime.get('seconds', 0),
                         caltype)
                 else:
-                    run_runtime = cpl_nml[cpl_group]['runtime']
+                    run_runtime = cpl_group['runtime']
 
-                # Now write out new run start date and total runtime.
-                cpl_nml[cpl_group]['inidate'] = cal.date_to_int(run_start_date)
-                cpl_nml[cpl_group][runtime0_key] = total_runtime
-                cpl_nml[cpl_group]['runtime'] = int(run_runtime)
+                # Now write out new run start date and total runtime into the
+                # work directory namelist.
+                cpl_group["init_date"] = cal.date_to_int(init_date)
+                cpl_group['inidate'] = cal.date_to_int(run_start_date)
+                cpl_group[model.runtime0_key] = previous_runtime
```

Comment:
When I test this, my `input_ice.nml` gave `    runtime0 = 3155673600.0`

The extra `.0` probably doesn't matter , but is it easy to tidy up?

Reply:
Good catch, I've adjusted the `seconds_between_dates` function to output integers:
```
return int(delta.total_seconds())
```
which should be safe as it's inputs are `datetime.date` objects, and so there shouldn't be any fractions of seconds appearing anywhere.

Reply:
nice